crates/cargo-bsp/src/server/global_state.rs

```diff
@@ -89,8 +89,9 @@ impl GlobalState {
         if let Some(response) = self.req_queue.incoming.cancel(request_id) {
             if let Some(handler) = self.handlers.get(&response.id) {
                 handler.cancel()
+            } else {
```

Comment:
Na pewno tak chcemy? Kiedy handlers.get() zwroci None?

Reply:
Funkcja cancel w global_state dotyczy wszystkich requestów, więc jeśli nie ma handlera (w którym jest wysyłanie response), to powinniśmy to zrobić tutaj